lib/seattleflu/db/cli/command/etl/manifest.py

```diff
@@ -88,48 +90,170 @@ def etl_manifest(*, action: str):
             with db.savepoint(f"manifest record {manifest_record.id}"):
                 LOG.info(f"Processing record {manifest_record.id}")
 
-                # Convert sample barcode to full identifier, ensuring it's
-                # known and from the correct identifier set.
-                sample_barcode = manifest_record.document.pop("sample")
-                sample_identifier = find_identifier(db, sample_barcode)
-
-                if not sample_identifier:
-                    LOG.warning(f"Skipping sample with unknown sample barcode «{sample_barcode}»")
-                    mark_skipped(db, manifest_record.id)
-                    continue
-
-                assert sample_identifier.set_name in expected_identifier_sets["samples"], \
-                    f"Sample identifier found in set «{sample_identifier.set_name}», not {expected_identifier_sets['samples']}"
-
-                # Optionally, convert the collection barcode to full
-                # identifier, ensuring it's known and from the correct
-                # identifier set.
-                collection_barcode = manifest_record.document.pop("collection", None)
-                collection_identifier = find_identifier(db, collection_barcode) if collection_barcode else None
-
-                if collection_barcode and not collection_identifier:
-                    LOG.warning(f"Skipping sample with unknown collection barcode «{collection_barcode}»")
-                    mark_skipped(db, manifest_record.id)
-                    continue
-
-                assert not collection_identifier \
-                    or collection_identifier.set_name in expected_identifier_sets["collections"], \
-                        f"Collection identifier found in set «{collection_identifier.set_name}», not {expected_identifier_sets['collections']}"
-
-                # Upsert sample cooperatively with enrollments ETL routine
-                #
-                # The details document was intentionally modified by two pop()s
-                # earlier to remove barcodes that were looked up.  The
-                # rationale is that we want just one clear place in the
-                # warehouse for each piece of information.
-                sample, status = upsert_sample(db,
-                    identifier            = sample_identifier.uuid,
-                    collection_identifier = collection_identifier.uuid if collection_identifier else None,
-                    additional_details    = manifest_record.document)
-
-                mark_loaded(db, manifest_record.id,
+                # Processing manifest records not involving self-test kits
+                if ("utm" not in manifest_record.document and 
+                        "rdt" not in manifest_record.document):
```

Comment:
I'm a little concerned about splitting the processing into two big conditional branches like this. Large, divergent conditional branches are an immediate red flag for me when writing/reading code. Effectively it means we have two separate ETL routines bundled into one. I think we could make it a unified routine, and if we can't, I think it'd be worth considering two separate routines.